notebooks/ribasim_delwaq/write_boundwq_files.py

```diff
@@ -0,0 +1,885 @@
+                ),
+            ),
+        )
+
+        df_delwaq_input["NH4"] = np.where(
+            df_pivot["NH4_0"] != -999,
+            df_pivot["NH4_0"],
+            np.where(
+                df_pivot["NH4_1"] != -999,
+                df_pivot["NH4_1"],
+                np.where(df_pivot["NH4_2"] != -999, df_pivot["NH4_2"], -999),
+            ),
+        )
+
+        df_delwaq_input["OON"] = np.where(
+            df_pivot["OON_0"] != -999,
+            df_pivot["OON_0"],
+            np.where(
+                df_pivot["OON_1"] != -999,
+                df_pivot["OON_1"],
+                np.where(df_pivot["OON_2"] != -999, df_pivot["OON_2"], -999),
+            ),
+        )
+
+        df_delwaq_input["PO4"] = np.where(
+            df_pivot["PO4_0"] != -999,
+            df_pivot["PO4_0"],
+            np.where(df_pivot["PO4_1"] != -999, df_pivot["PO4_1"], -999),
+        )
+
+        df_delwaq_input["AAP"] = np.where(
+            df_pivot["AAP_0"] != -999,
+            df_pivot["AAP_0"],
+            np.where(df_pivot["AAP_1"] != -999, df_pivot["AAP_1"], -999),
+        )
+
+        df_delwaq_input["OOP"] = np.where(
+            df_pivot["OOP_0"] != -999,
+            df_pivot["OOP_0"],
+            np.where(df_pivot["OOP_1"] != -999, df_pivot["OOP_1"], -999),
+        )
+
+        # Only include if we decided to include Ntot and Ptot and CHLFa
+        # df_delwaq_input['Ntot'] = np.where(df_pivot['Ntot_0'] != -999, df_pivot['Ntot_0'],
+        # np.where(df_pivot['Ntot_1'] != -999, df_pivot['Ntot_1'], -999))
+        # df_delwaq_input['Ptot'] = np.where(df_pivot['Ptot_0'] != -999, df_pivot['Ptot_0'],
+        #                         np.where(df_pivot['Ptot_1'] != -999, df_pivot['Ptot_1'], -999))
+        # df_delwaq_input['CHLFa'] = df_pivot['CHLFa'] if 'CHLFa' in df_pivot else -999
+
+        dict_delwaq_input[loc] = df_delwaq_input
+    return dict_choosen_method, dict_delwaq_input
+
+
+# Converteer de IM-data voor buitenlandse aanvoeren
+dict_choosen_method_IM, dict_delwaq_input_IM = process_measurement_data(df_IM_metingen, locations_IM, parameters_IM)
+
+# Converteer de Z-info data voor RWZIs
+dict_choosen_method_Zinfo, dict_delwaq_input_Zinfo = process_measurement_data(
+    df_Zinfo, locations_zinfo, parameters_Zinfo
+)
+
+# Log the parameters
+parameter_delwaq_input_IM = list(dict_delwaq_input_IM.values())[0].columns[2:]
+parameter_delwaq_input_Zinfo = list(dict_delwaq_input_Zinfo.values())[0].columns[2:]
+
+logging.info(
+    f"Delwaq input parameters uit IM-metingen (buitenlandse aanvoeren): \n {list(parameter_delwaq_input_IM[:])} \n voor {len(dict_delwaq_input_IM.keys())} locaties"
+)
+logging.info(
+    f"Delwaq input parameters uit Z-Info (rwzis): \n {list(parameter_delwaq_input_Zinfo[:])} \n voor {len(dict_delwaq_input_Zinfo.keys())} locaties"
+)
+
+
+# %% Schijf de boundarywq.dat files weg
+def write_boundwq_file(boundwq_file, dict_delwaq_input, parameter_delwaq_input):
```

Comment:
To make this easy to read and use, I generally prefer to add a short docstring (in English) and to add type annotations, see for example annotations:

https://github.com/Deltares/Ribasim-NL/blob/486b5817ab7c50535d520041ad0038ebb5838c98/notebooks/samenvoegen_modellen.py#L215-L223